build_tools/cmake/rebuild.sh

```diff
@@ -13,6 +13,8 @@
 # limitations under the License.
 
 # Build the IREE project with CMake. Designed for CI, but can be run manually.
+# This uses previously cached build results and does not clear the build
+# directory. For a clean build, use build_tools/cmake/clean_build.sh
```

Comment:
Are there enough logs between `set -x`, the build environments, and CMake itself here? Want to include something explicit on whether there are or are not cached results?

Reply:
Good idea. Added logging.